### PR TITLE
Allowing filename to be empty in json settings file for solc. Fixing …

### LINKED ISSUE #8371: ICE in solfuzzer; file is invalid to solc
## Description

This contract:
```solidity
import {A} from ".";
```

produces:

```
Testing compiler with optimizer.
Input JSON: {"language":"Solidity","settings":{"evmVersion":"homestead","optimizer":{"enabled":true,"runs":200},"outputSelection":{"*":{"":["*"],"*":["*"]}}},"sources":{"":{"content":"import {A} from \".\";\n"}}}
Output JSON: {"errors":[{"component":"general","formattedMessage":"InternalCompilerError: \n","message":"Internal compiler error (/home/user/solidity/libsolidity/interface/CompilerStack.cpp:1001):","severity":"error","type":"InternalCompilerError"}],"sources":{}}
Invalid error: "InternalCompilerError"
Fuzzer failed on finish.sol
```

when compiled with solfuzzer.  solc just complains about the import.

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer, now with a few more mutation operators

## Environment

- Compiler version:  0.6.3-develop.2020.2.10+commit.e8eb1f2d.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

### PR DESCRIPTION
**Pull request by mijovic**
Fixes #8371

### Checklist
- [ ] Code compiles correctly
- [ ] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [ ] Used meaningful commit messages

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Compiler Features:
 
 Bugfixes:
  * isoltest: Added new keyword `wei` to express function value in semantic tests
+ * solc: Filename can be empty in case of standard-json solc run
```
Comment (chriseth) [inline Changelog.md]:
```suggestion
 * Standard-JSON-Interface: Fix a bug related to empty filenames and imports.
```

**Comment by axic** [comment]
Is it a good idea to allow empty names? Did anyone check all the outputs we produce whether it works with that or not?

**Comment by leonardoalt** [comment]
@axic I thought the same, especially regarding output selection and configuration. Tests pass though, so it's probably fine.

**Comment by chriseth** [comment]
In standard-json, I don't think it is a problem. We allowed empty inputs before, the assert was just in relation to imports.

**Comment by chriseth** [comment]
Also note that we use empty file names all the time in tests.

**Review by chriseth — APPROVED** [review]

**Pull request merged**